Week5/Blockbuster/app/src/main/java/com/example/blockbuster/fragments/MainFragment.kt

```diff
@@ -0,0 +1,140 @@
+package com.example.blockbuster.fragments
+
+import android.content.DialogInterface
+import android.net.Uri
+import android.os.Bundle
+import android.view.*
+import androidx.activity.OnBackPressedCallback
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.GridLayoutManager
+import com.example.blockbuster.R
+import com.example.blockbuster.data.DataManager
+import com.example.blockbuster.data.Movie
+import com.example.blockbuster.recyclerview.MovieAdapter
+import kotlinx.android.synthetic.main.fragment_main.*
+import kotlinx.android.synthetic.main.switch_layout.view.*
+
+
+class MainFragment : Fragment() , MovieAdapter.MovieClickListener {
+
+    private lateinit var model : DataManager
+    private val adapter = MovieAdapter(mutableListOf(),this)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        // -------------------------------------Show toolbar
+        val activity = activity as AppCompatActivity
+        activity.supportActionBar?.show()
+        // -------------------------------------------------
+        // -------------------------------------------------------Initialize Movie List
+        model = activity.let { ViewModelProvider(it).get(DataManager::class.java) }
+        model.initMovieList()
```

Comment:
Nice job initializing the movies and on creating the room database and repository, the overall architecture works.